qhana_plugin_runner/db/models/tasks.py

```diff
@@ -69,7 +69,7 @@ class TaskData:
     Attributes:
         id (int): ID of corresponding :class:`ProcessingTask` entry. Use the id to fetch this information from the database.
         key (str): a key in dict
-        value (str, optional): a corresponding value in dict
+        value (Union[dict, list, str, float, int, bool, None], optional): a corresponding value in dict. Note that SQL Alchemy's JSON type is used which does not support in-place mutations. If a value needs to be changed, e.g., use `copy.deepcopy`.
```

Comment:
We should test if an actual deepcopy is needed here before advising users to always use deepcopy. It should not matter much for performance as it will probably only be used a few times per plugin call but it just does not feel right for me.